### ISSUE TITLE
Question about cross-vocabulary distillation setup

### ISSUE THREAD

**Issue by e8393373-rgb** [UNITES-Lab/DOGe#5]
Hello, thanks for the awesome repo!

I've got a question about the code in src/doge/models/distill_wrapper.py:

`lm_loss = teacher_outputs.loss
teacher_logits = teacher_outputs.logits[..., :self.vocab_size]
proxy_logits = proxy_outputs.logits[..., :self.vocab_size]

teacher_logits = teacher_logits.to(torch.float32)
proxy_logits = proxy_logits.to(torch.float32)`

Why are you clipping the logits by vocab size? In your paper, the Teacher model (DeepSeek-R1-Distill-Qwen-7B) and Student model (Llama-3.2-1B) have completely different vocabulary sizes, right? I'm thinking this could be an issue since the token IDs at the same positions don't actually correspond to the same tokens across the two vocabularies. Wouldn't this mess up the KL divergence calculation and make it hard to reproduce your results?
Could you walk me through the logic here? Am I missing something about how you handle the vocab mismatch?

**Comment by pingzhili** [comment]
thanks for your question. the code you shared here is for defensive training, where all teachers and students always share the same vocab. we do not have the `DeepSeek-R1-Distill-Qwen-7B` + `Llama-3.2-1B` setting during defensive training.

I belive the `DeepSeek-R1-Distill-Qwen-7B` + `Llama-3.2-1B` setting you mentioned here is from Figure 3, which refers to the normal text-level distillation as an evaluation of our defensive training. during normal text-level distillation, we only apply text-level distillation, where there is no KL loss on logits.

**Comment by pingzhili** [comment]
p.s.

if you have any specific experimental results that you believe are not consistent with ours, please share them here—we’re happy to discuss them.

otherwise, please don't use such a misleading title, especially with a brand-new github account. it just looks suspicious and can leave me the impression of intentionally hiding something. (or it is an honor for us that you joined github only for our repo)

**Comment by e8393373-rgb** [comment]
I've updated the title to better reflect my actual question. 
Thanks for clarifying the distinction between defensive training and evaluation!

I understand now that the code provided is for defensive training (same vocab), 
but I'm curious about the evaluation setup in Figure 3 where you test 
DeepSeek-R1-Distill-Qwen-7B → Llama-3.2-1B distillation.

Could you share:
1. How did you perform the text-level distillation for the cross-vocab evaluation?
2. If possible, could you point me to the distillation script used for Figure 3?

I'd love to reproduce those results. Thanks!

**Comment by pingzhili** [comment]
1. there are two steps. the first is to generate the response text data on a specific prompt set from the teacher model, as shown in https://github.com/UNITES-Lab/DOGe/blob/main/generate.sh; the second is to use the generated response to finetune the studen model, as shown in https://github.com/UNITES-Lab/DOGe/blob/main/train-distill.sh 

2. you may want to check the readme and follow the specified steps; you will need to modify task/model accordingly for specific setting

**Comment by e8393373-rgb** [comment]
Thanks a lot!
